### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this feature me and my partner did the following

- Used git flow
- Made sure to use git flow
- Followed Ruby best practices
- Used Rspec for unit tests
- Created spec folder containing our unit test files
- Made sure that all tests run well

### CODE REVIEW

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @bhushan354 and @Chimoneg27, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Tests are passing for the `class` spec added inside the `spec` folder ✅ 

## Required Changes ♻️
- [ ] Please adhere to the `project` **requirements**, as per the instructions you should write tests for all the classes including the decorator but it seems like you wrote the `tests` for the four classes, kindly write the test for `classroom`, `renta;s` and all decorator classes that you implemented previously.

Screenshot from 2023-11-17 00-54-21

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class/spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      @book.title.should eql 'title'
+    end
+  end
+
+  describe '#author' do
+    it 'returns the correct author' do
+      @book.author.should eql 'author'
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline class/spec/book_spec.rb]: - [ ] please remember to test all methods in a class, currently you did not add unit test for the method `add_rental` kindly include that

File: class/spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      expect(@person.name).to eql 'maximilianus'
+    end
+  end
+
+  describe '#age' do
+    it 'returns the correct age' do
+      expect(@person.age).to eql 22
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline class/spec/person_spec.rb]: - [ ] Please remember to test all methods in a class, currently, you did not add unit test for the method `correct_name` kindly include that

File: class/spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      it 'returns a student object' do
+        expect(@student).to be_an_instance_of Student
+      end
+
+      it 'returns the students name' do
+        expect(@student.name).to eql 'Unknown'
+      end
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline class/spec/student_spec.rb]: - [ ] Please remember to test all methods in a class, currently, you did not add a unit test for the method `play_hooky` kindly include that

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Chimoneg27 , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class/spec/student_spec.rb (hunk 3 above)
Comment (Chimoneg27) [reply class/spec/student_spec.rb]: Are these the only changes

Comment (VICTOR-CHUKWU) [reply class/spec/student_spec.rb]: yes please

File: class/spec/book_spec.rb (hunk 1 above)
Comment (peterrobert) [reply class/spec/book_spec.rb]: This  test has been added as requested.

Comment (peterrobert) [reply class/spec/book_spec.rb]: This test has been added well

File: class/spec/person_spec.rb (hunk 2 above)
Comment (peterrobert) [reply class/spec/person_spec.rb]: This test has been added.

File: class/spec/student_spec.rb (hunk 3 above)
Comment (peterrobert) [reply class/spec/student_spec.rb]: This test has been added well

**Review by peterrobert — APPROVED** [review]
Hi @Chimoneg27, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Awesome work making the changes that were requested by the previous reviewer. The tests are all passing✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**